markdown/events/2023/bytesize_precommit.md

```diff
@@ -23,3 +23,39 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[3:35](https://www.youtube.com/watch?v=08d6zv6zvdM&t=215)
+One of the nice things with pre-commit is that you can use every code linting tool you want. Also, it doesn't matter if it's in a different language. Like you see with prettier, which is actually an NPM tool. We don't need to have node installed to run this version of prettier with pre-commit. It just comes through the mirror there. But other tools like Python-based tools come directly from the tools itself. If we set this config up like here, like we have for the tools repository, for example, it automatically checks the Python files with Black and isort. In this example, I added again to the Ruby file, but I also switched the import statements in our main.py. If I then run commit, it not only runs prettier, but it also runs Black and isort. Black was satisfied, isort found that there was an error there and fixed it fast. Prettier fixed it fast as well within the README. The only changes then were in the README, because we already had it nicely sorted before. Hit the second commit. Now the code is nice again. That was pretty much it.
+
+[5:05](https://www.youtube.com/watch?v=08d6zv6zvdM&t=305)
+Just a quick shout out to the person who actually brought this tool to us, which was Fabian. It was more than an idea how to always have prettier available in tools without requiring people to install node. He found this tool. Actually it's very nice with having it also run with pipelines and everything. Praise Fabian. With that, I'm open for any questions if there are.
+
+[5:38](https://www.youtube.com/watch?v=08d6zv6zvdM&t=338)
+(host) Thank you very much. If you have any questions, then you can either write them in the chat, or you can just ask them straight away. Everyone should be now able to unmute themselves. Do we have any questions? It doesn't seem so. Maxime. Maxime says hello or has a question?
+
+(question) Yes, I have a question. You said that it's in tools. Is it already released, or is it in the coming release.
+
+(answer) In tools, it's already in the release, because actually we run Prettier with pre-commit whenever we dump YAML files or JSON files in tools. For example, when we create the tests for modules, these files are now prettified with Prettier, because before we had the problem that our function actually dumped code, Prettier didn't like. Now we run pre-commit to run Prettier on it. Also itself, the repository has Black and isort. There, it's already in. But for the pipelines, the next release will have it in a template. Then all the pipelines also get it. You just need to run `pre-commit install` to activate it. Modules since yesterday got the got the config in. If you now are in module. Pull. Then run `pre-commit install`, all your modules will automatically run Prettier. Or like all changes on modules will automatically be run through Prettier before you can commit. I recommend to do that. If you write modules or subworkflows.
```

Comment:
```suggestion
(answer) In tools, it's already in the release, because actually we run Prettier with pre-commit whenever we dump YAML files or JSON files in tools. For example, when we create the tests for modules, these files are now prettified with Prettier, because before we had the problem that our function actually dumped code, Prettier didn't like. Now we run pre-commit to run Prettier on it. Also the repository itslef has Black and isort. There, it's already in. But for the pipelines, the next release will have it in a template. Then all the pipelines also get it. You just need to run `pre-commit install` to activate it. Modules since yesterday got the got the config in. If you now are in module. Pull. Then run `pre-commit install`, all your modules will automatically run Prettier. Or all changes on modules will automatically be run through Prettier before you can commit. I recommend to do that. If you write modules or subworkflows.
```